src/osal/lv_freertos.c

```diff
@@ -260,13 +266,24 @@ lv_result_t lv_thread_sync_wait(lv_thread_sync_t * pxCond)
 
 lv_result_t lv_thread_sync_signal(lv_thread_sync_t * pxCond)
 {
-#if USE_FREERTOS_TASK_NOTIFY
-    /* Send a notification to the task waiting. */
-    xTaskNotifyGive(pxCond->xTaskToNotify);
-#else
     /* If the cond is uninitialized, perform initialization. */
     prvCheckCondInit(pxCond);
 
+#if USE_FREERTOS_TASK_NOTIFY
+    _enter_critical();
```

Comment:
I see this like:
```
xTaskNotifyGive(pxCond->xTaskToNotify);
pxCond->xSyncSignal = pdTRUE;
```

Reply:
I mean to replace all the code under `#if USE_FREERTOS_TASK_NOTIFY` with the above suggestion and see if it works to you.

Reply:
> Replace all new code with:
> 
> ```
> #if USE_FREERTOS_TASK_NOTIFY
>     xTaskNotifyGive(pxCond->xTaskToNotify);
>     pxCond->xSyncSignal = pdTRUE;
> #else
> ```

`pxCond->xTaskToNotify` may be `NULL` which is why there is the current logic to determine which task to notify, if any.